src/utility-types.ts

```diff
@@ -37,3 +38,21 @@ export interface MessageValue<
    */
   readonly parent?: MainWorker
 }
+
+/**
+ * TODO: Needs description.
+ *
+ * @template Worker Type of worker.
+ * @template Response Type of response of execution. This can only be serializable data.
+ */
+export interface ThisNeedsAName<Worker extends IWorker, Response = unknown> {
+  /**
+   * Resolve callback to fulfill the promise.
+   */
+  readonly resolve: (value: Response) => void
+  /**
+   * Reject callback to reject the promise.
+   */
+  readonly reject: (reason?: string) => void
+  readonly worker: Worker
```

Comment:
We should add JSDoc to this property

Reply:
To keep It simple I would descrive It as the worker that has the task assigned